app/assets/javascripts/utilities/userData.js

```diff
@@ -3,5 +3,11 @@
 function userData() {
   const { user = null } = document.body.dataset;
 
+  // [@forem/oss]: there's an edge case in which `user` has the value of "undefined",
+  // this results in is a JSON syntax error (`user` should be a hash, not a string).
+  if (user === 'undefined') {
```

Comment:
Odd that it is a string `'undefined'`. The default value of `null` handles the case when user is `undefined` so that `JSON.parse(user)` works all the time.

It'd be good to know why it's a  `'undefined'` instead of actually `undefined`